rotkehlchen/api/rest.py

```diff
@@ -4675,15 +4680,50 @@ def export_history_events(
             )
 
         if len(history_events) == 0:
-            return api_response(
-                wrap_in_fail_result('No history processed in order to perform an export'),
+            return wrap_in_fail_result(
+                message='No history processed in order to perform an export',
                 status_code=HTTPStatus.CONFLICT,
             )
 
+        with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
+            settings = self.rotkehlchen.get_settings(cursor)
+            currency = settings.main_currency.resolve_to_asset_with_oracles()
+
         serialized_history_events = []
         headers: dict[str, None] = {}
         for event in history_events:
-            serialized_event = event.serialize_for_csv()
+            if event.balance.usd_value != ZERO and currency != A_USD:
```

Comment:
I'm not sure about the condition here. What happens if the event has `usd_value == ZERO and currency == A_USD`. In that case you won't enter this section and the price will be exported as 0 right?

Reply:
right, i had the wrong assumption that `usd_value` is always fetched with the event.